TYPO3.Neos/Classes/TYPO3/Neos/Domain/Service/NodeSearchService.php

```diff
@@ -62,8 +77,16 @@ public function findByProperties($term, array $searchNodeTypes, Context $context
         $searchResult = array();
         $nodeTypeFilter = implode(',', $searchNodeTypes);
         $nodeDataRecords = $this->nodeDataRepository->findByProperties($term, $nodeTypeFilter, $context->getWorkspace(), $context->getDimensions(), $startingPoint ? $startingPoint->getPath() : null);
+        $baseContextProperties = $context->getProperties();
+        /** @var NodeData $nodeData */
         foreach ($nodeDataRecords as $nodeData) {
-            $node = $this->nodeFactory->createFromNodeData($nodeData, $context);
+            $site = $this->siteRepository->findOneByNodePath($nodeData->getPath());
+            $contextProperties = array_merge($baseContextProperties, [
+                'currentSite' => $site,
+                'currentDomain' => $site->getFirstActiveDomain()
+            ]);
```

Comment:
this doesn't really belong here, while cleaning up this we should get rid of all those places we have this similar logic..